### PR TITLE
ADCM-2362 Forbid duplicated names in config

### PR DESCRIPTION
**Pull request by DanBalalan**

### CODE REVIEW

File: python/cm/stack.py [hunk 1]
```diff
@@ -516,6 +517,27 @@ def get_yspec(proto, ref, bundle_hash, conf, name, subname):
     return schema
 
 
+def check_unique_names_constraint(conf_dict):
+    names = defaultdict(list)
+    firs_level_key = 'firs_level_key'
```
Comment (kuhella) [inline python/cm/stack.py:522]: Typo: first_level_key.
maybe value to 'root'?

File: python/cm/stack.py [hunk 2]
```diff
@@ -516,6 +517,27 @@ def get_yspec(proto, ref, bundle_hash, conf, name, subname):
...
+            if is_group(conf):
+                for subconf in conf['subs']:
+                    names[conf['name']].append(subconf['name'])
+    else:
+        raise NotImplementedError
+    for key in names:
+        duplicated = [i for i in names[key] if names[key].count(i) > 1]
+        if duplicated:
+            msg = f'Duplicated names in {key} section: {", ".join(i for i in set(duplicated))}'
+            err('DUPLICATED_NAME_IN_CONFIG', msg)
```
Comment (kuhella) [inline python/cm/stack.py:538]: Also it would be nice to add to log that info

**Review by kuhella — CHANGES_REQUESTED** [review]

File: python/cm/stack.py (hunk 1 above)
Comment (DanBalalan) [reply python/cm/stack.py]: done

**Comment by kuhella** [comment]
This branch is outdated and brokes backward compatibility

**Comment by kuhella** [comment]
This branch is outdated and brokes backward compatibility

**Comment by kuhella** [comment]
This branch is outdated and brokes backward compatibility

**Comment by kuhella** [comment]
This branch is outdated and brokes backward compatibility

**Pull request closed without merging**